add feature collection layer from table sample (#415)
Esri/arcgis-maps-sdk-kotlin-samples: samples/add-feature-collection-layer-from-table/README.md

# Add feature collection layer from table

Create a Feature Collection Layer from a Feature Collection Table, and add it to a map.

![Image of add feature collection layer from table](add-feature-collection-layer-from-table.png)

## Use case

A Feature Collection allows easily importing external data (such as CSV files), as well as creating custom schema for data that is in non-standardized format. This data can then be used to populate a Feature Collection Table, and displayed in a Feature Collection Layer using the attributes and geometries provided in the external data source. For example, an electricity supplier could use this functionality to visualize existing location data of coverage areas (polygons), power stations (points), transmission lines (polylines), and others.

## How to use the sample

When launched, this sample displays a `FeatureCollectionLayer` with a `Point`, `Polyline` and `Polygon` geometry. Pan and zoom to explore the scene.

## How it works

1. Create a `FeatureCollectionLayer` using a new feature collection, `FeatureCollectionLayer(featureCollection)`

> 2. Add the feature collection layer to the map, `ArcGISMap.getOperationalLayers().add(featureCollectionLayer)`.

Reviewer: ```suggestion
2. Add the feature collection layer to the map, `ArcGISMap.operationalLayers.add(featureCollectionLayer)`.
```

> 3. Create a `FeatureCollectionTable` for the `GeometryType`s `Point`, `Polyline`, and `Polygon`, `FeatureCollectionTable(fields, geometryType, spatialRefernce)`

Reviewer: ```suggestion
3. Create a `FeatureCollectionTable` for the `GeometryType`s `Point`, `Polyline`, and `Polygon`, `FeatureCollectionTable(fields, geometryType, spatialReference)`
```

4. Additionally, pass in a list of `Field` objects to represent the table's schema. In this case a field of type String named `name` is added.
5. Assign a `SimpleRenderer` to each table to render any `Feature`s from that table using the `Symbol` that was set.

> 6. Add the feature collection table to the feature collection, `FeatureCollection.getTables().add(featureCollectionTable)`.

Reviewer: ```suggestion
6. Add the all feature collection tables to the feature collection, `FeatureCollection(featureCollectionTables = listOf(points, lines, polygons))`.
```

> 6. Add the all feature collection tables to the feature collection, `FeatureCollection(featureCollectionTables = listOf(points, lines, polygons))`.

Reviewer: This was due to a typo in my feedback: 

```suggestion
6. Add all the feature collection tables to the feature collection, `FeatureCollection(featureCollectionTables = listOf(points, lines, polygons))`.
```

Author: addressed

7. Use the `createFeature` method to create a feature from the feature collection table, passing an attribute and geometry for that feature, `FeatureCollectionTable.createFeature(attributes, geometry)`.

> 8. Add new features to the table, `FeatureCollectionTable.addFeatureAsync(feature)`.

Reviewer: ```suggestion
8. Add new features to the table, `FeatureCollectionTable.addFeature(feature)`.
```
